src/ccnl-pkt/src/ccnl-pkt-iottlv.c

```diff
@@ -80,18 +80,28 @@ ccnl_iottlv_peekType(unsigned char *buf, int len)
     return typ;
 }
 
-// parse TL (returned in typ and vallen) and adjust buf and len
+/**
+ * Opens a TLV and reads the Type and the Length Value
+ * @param buf allocated buffer in which the tlv should be opened
+ * @param len length of the buffer
+ * @param typ return value via pointer: type value of the tlv
+ * @param vallen return value via pointer: length value of the tlv
+ * @return 0 on success, -1 on failure.
+ */
 int
 ccnl_iottlv_dehead(unsigned char **buf, int *len,
                    unsigned int *typ, int *vallen)
 {
+    size_t maxlen = *len;
```

Comment:
Again, same remark on len and negative values (like before)

Reply:
ok, on another position this is used as status code. so i cannot make unsigned in a simple way. nevertheless, the function ensures, len value is at least 1. otherwise packet will be dropped. so this is not an issue here.